include/libdnf/conf/config_main.hpp

```diff
@@ -0,0 +1,187 @@
+    OptionBool & debug_solver();
+    OptionStringList & installonlypkgs();
+    OptionStringList & group_package_types();
+
+    /*  NOTE: If you set this to 2, then because it keeps the current
+    kernel it means if you ever install an "old" kernel it'll get rid
+    of the newest one so you probably want to use 3 as a minimum
+    ... if you turn it on. */
+    OptionNumber<std::uint32_t> & installonly_limit();
+
+    OptionStringList & tsflags();
+    OptionBool & assumeyes();
+    OptionBool & assumeno();
+    OptionBool & check_config_file_age();
+    OptionBool & defaultyes();
+    OptionBool & diskspacecheck();
+    OptionBool & localpkg_gpgcheck();
+    OptionBool & gpgkey_dns_verification();
+    OptionBool & obsoletes();
+    OptionBool & showdupesfromrepos();
+    OptionBool & exit_on_lock();
+    OptionSeconds & metadata_timer_sync();
+    OptionStringList & disable_excludes();
+    OptionEnum<std::string> & multilib_policy();  // :api
+    OptionBool & best();                          // :api
+    OptionBool & install_weak_deps();
+    OptionString & bugtracker_url();
+    OptionBool & zchunk();
+    OptionEnum<std::string> & color();
+    OptionString & color_list_installed_older();
+    OptionString & color_list_installed_newer();
+    OptionString & color_list_installed_reinstall();
+    OptionString & color_list_installed_extra();
+    OptionString & color_list_available_upgrade();
+    OptionString & color_list_available_downgrade();
+    OptionString & color_list_available_reinstall();
+    OptionString & color_list_available_install();
+    OptionString & color_update_installed();
+    OptionString & color_update_local();
+    OptionString & color_update_remote();
+    OptionString & color_search_match();
+    OptionBool & history_record();
+    OptionStringList & history_record_packages();
+    OptionString & rpmverbosity();
+    OptionBool & strict();                    // :api
+    OptionBool & skip_broken();               // :yum-compatibility
+    OptionBool & autocheck_running_kernel();  // :yum-compatibility
+    OptionBool & clean_requirements_on_remove();
+    OptionEnum<std::string> & history_list_view();
+    OptionBool & upgrade_group_objects_upgrade();
+    OptionPath & destdir();
+    OptionString & comment();
+    OptionBool & downloadonly();
+    OptionBool & ignorearch();
+
+    OptionString & module_platform_id();
+    OptionString & user_agent();
+    OptionBool & countme();
+
+    // Repo main config
+    OptionNumber<std::uint32_t> & retries();
+    OptionString & cachedir();
+    OptionBool & fastestmirror();
+    OptionStringList & excludepkgs();
+    OptionStringList & includepkgs();
+    OptionString & proxy();
+    OptionString & proxy_username();
+    OptionString & proxy_password();
+    OptionEnum<std::string> & proxy_auth_method();
+    OptionStringList & protected_packages();
+    OptionString & username();
+    OptionString & password();
+    OptionBool & gpgcheck();
```

Comment:
Can we go ahead and lay the groundwork here for the GPG check refactor? (As noted in [RhBug:1433592#c7](https://bugzilla.redhat.com/show_bug.cgi?id=1433592#c7)). This is the perfect time to do so.

Reply:
@Conan-Kudo Am I correct that this is a request to drop RHEL7, RHEL8 compatibility in favour of Zypper behaviour?

Reply:
Somewhat. The `gpgcheck_policy` option would control whether we behave as YUM does or as Zypper does for `gpgcheck`. In general, I think the correct default is to behave like Zypper here, but I'm okay with making the default the same as YUM, as long as we can tune this.